File: update_constants.js

const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'lib/constants.ts');

try {
    let content = fs.readFileSync(filePath, 'utf8');

    // Regex to match each product object in the PRODUCT_CATEGORIES array
    // We look for objects dealing with PRODUCT_CATEGORIES
    // Heuristic: find the array definition, then iterate inside it.

    // Easier approach: replace `slug: "some-slug",` with `metaTitle: "...", metaDescription: "...", slug: "some-slug",`
    // but we need the 'name' field to generate the title.

    // Let's iterate over matches.

    // Regex explanation:
    // name: "(?<name>.*?)"
    // ... (any content until slug)
    // slug: "(?<slug>.*?)"

    const regex = /name:\s*"(.*?)",[\s\S]*?(slug:\s*".*?",)/g;

    const newContent = content.replace(regex, (match, name, slugPart) => {
        if (match.includes('metaTitle:')) return match; // Already updated

        const metaTitle = `${name} | Axis Packaging`;
        const metaDescription = `Custom ${name.toLowerCase()} tailored to your brand. High-quality, sustainable packaging solutions available at Axis Packaging.`;

        // Insert before the slug part
        // We want to keep the indentation of the slug line mostly, or just add new lines.
        // The match includes everything from name... to slug...
        // We can replace the `slug:` part with `meta... slug:`

        return match.replace(slugPart, `metaTitle: "${metaTitle}",\n    metaDescription: "${metaDescription}",\n    ${slugPart}`);
    });

    fs.writeFileSync(filePath, newContent, 'utf8');
    console.log('Successfully updated lib/constants.ts');
} catch (err) {
    console.error('Error updating file:', err);
}

File: update_constants.py

import re

file_path = 'lib/constants.ts'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

def replacer(match):
    block = match.group(0)
    # Extract name
    name_match = re.search(r'name: "(.*?)",', block)
    if not name_match:
        return block
    
    name = name_match.group(1)
    
    # Check if metaTitle already exists to avoid double addition (idempotency)
    if 'metaTitle:' in block:
        return block

    # Generate meta fields
    meta_title = f'{name} | Axis Packaging'
    meta_desc = f'Custom {name.lower()} tailored to your brand. High-quality, sustainable packaging solutions available at Axis Packaging.'
    
    # Insert before slug (or at the end of the object)
    # The pattern matches the whole object content, let's inject before "slug:"
    if 'slug:' in block:
        return block.replace('slug:', f'metaTitle: "{meta_title}",\n    metaDescription: "{meta_desc}",\n    slug:')
    else:
        return block

# Pattern to match each object in the array
# We assume the structure is consistent as seen in the view_file output
# {
#   name: "...",
#   ...
#   slug: "...",
# }
# We'll match from { to }, non-greedy? 
# The objects are simple enough.
pattern = re.compile(r'\{\s+name:.*?\s+slug:.*?\s+\},', re.DOTALL)

new_content = pattern.sub(replacer, content)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Updated lib/constants.ts")
